lms/djangoapps/courseware/views/views.py

```diff
@@ -929,47 +929,54 @@ def _get_cert_data(student, course, course_key, is_active, enrollment_mode):
             _('Your enrollment: Audit track'),
             _('You are enrolled in the audit track for this course. The audit track does not include a certificate.'),
             download_url=None,
-            cert_web_view_url=None,
-            may_view_certificate=None
+            cert_web_view_url=None
         )
 
-    show_message = (
-        is_active and CourseMode.is_eligible_for_certificate(enrollment_mode)
-        and certs_api.cert_generation_enabled(course_key)
-    )
+    may_view_certificate = False
+    # https://openedx.atlassian.net/browse/EDUCATOR-1204: historically,
+    # certificates for self-paced courses are displayed no matter the
+    # state of advanced course settings regarding certificates.
+    if course_key:
+        course = get_course_by_id(course_key)
+        may_view_certificate = course.self_paced or course.may_certify()
+
+    show_message = all([
+        is_active,
+        CourseMode.is_eligible_for_certificate(enrollment_mode),
+        certs_api.cert_generation_enabled(course_key),
+        may_view_certificate
+    ])
 
     if not show_message:
         return None
 
-    may_view_certificate = course_key and get_course_by_id(course_key).may_certify()
-
     if certs_api.is_certificate_invalid(student, course_key):
         return CertData(
             CertificateStatuses.invalidated,
             _('Your certificate has been invalidated'),
             _('Please contact your course team if you have any questions.'),
             download_url=None,
-            cert_web_view_url=None,
-            may_view_certificate=None
+            cert_web_view_url=None
         )
 
     cert_downloadable_status = certs_api.certificate_downloadable_status(student, course_key)
 
     if cert_downloadable_status['is_downloadable']:
         cert_status = CertificateStatuses.downloadable
         title = _('Your certificate is available')
-        msg = _('You can keep working for a higher grade, or request your certificate now.')
+        msg = _("You've earned a certificate for this course.")
         if certs_api.has_html_certificates_enabled(course_key, course):
```

Comment:
@iloveagent57 What about "You can view your certificate on the learner dashboard" or something similar?